### ISSUE TITLE
[RFE] symbol versionning

### ISSUE THREAD

**Issue by remicollet** [edenhill/librdkafka#1293]
Hi,

I notice librdkafka use symbol visibility/versionning using ` --version-script=librdkafka.lds`

But sadly, all symbols are in the "global" section.

Will be nice to have symbols listed per version
Ex, from http://rpms.remirepo.net/compat_reports/librdkafka/0.9.5_to_0.11.0-RC2/compat_report.html
Upcomming version 0.11 introduce 15 new symbols.

Example for libxml2 => https://git.gnome.org/browse/libxml2/tree/libxml2.syms

This will be useful for downstream distribution, as this can be auto-translated to dependencies (at least RPM does this)

```
$ rpm -q --requires php-xml
libxml2.so.2()(64bit)
libxml2.so.2(LIBXML2_2.4.30)(64bit)
libxml2.so.2(LIBXML2_2.5.0)(64bit)
libxml2.so.2(LIBXML2_2.5.2)(64bit)
libxml2.so.2(LIBXML2_2.5.4)(64bit)
...
$ rpm -q --requires php-pecl-rdkafka afka
librdkafka.so.1()(64bit)

```

So this ensure application using librdkafka (ex pecl/librdkafka) won't be installed without the proper version (soname is not enough, ok for dropped symbols, when soname is bump, not for added symbols, when soname doesn't change)

For now, the workaround is to ensure runtime version >= buildtime version using a manually added dependency (in the spec file).

```
%global buildver %(pkg-config --silence-errors --modversion rdkafka 2>/dev/null || echo 65536)
Requires:       librdkafka%{?_isa}  >= %{buildver}
```

**Comment by edenhill** [comment]
That seems reasonable but there's unfortunately not enough time to fix it for the upcoming 0.11.0 release.

**Comment by edenhill** [comment]
@remicollet Are there any compatibility concerns with existing applications when moving the existing symbols from global to their appropriate versions?
If not, we could (retroactively) annotate each public API with its release version (which is also useful from a documentation perspective) and then have the lds-gen.py script use that information to put the symbols in the correct section.
